src/http/httpd.cc

```diff
@@ -492,6 +492,10 @@ future<> http_server::listen(socket_address addr) {
     return listen(addr, lo);
 }
 future<> http_server::stop() {
+    if (_tasks_done.has_value()) {
+        _kill_timer.cancel();
```

Comment:
This doesn't look right, or maybe I'm missing something?
Since graceful_pre_stop() doesn't wait for anything, the user will immediately call stop(), and at that point the timer appears to be cancelled. But then, the tasks may never finish because what the timer was supposed to do (close all the active connections after the timeout) will never happen!

Reply:
That was a bug. I updated the logic a bit, now it uses with_timeout and graceful_pre_stop waits for it. So no timer needed.